pkg/scalers/datadog_scaler.go

```diff
@@ -389,15 +644,36 @@ func (s *datadogScaler) GetMetricSpecForScaling(context.Context) []v2.MetricSpec
 
 // GetMetricsAndActivity returns value for a supported metric and an error if there is a problem getting the metric
 func (s *datadogScaler) GetMetricsAndActivity(ctx context.Context, metricName string) ([]external_metrics.ExternalMetricValue, bool, error) {
-	num, err := s.getQueryResult(ctx)
-	if err != nil {
-		s.logger.Error(err, "error getting metrics from Datadog")
-		return []external_metrics.ExternalMetricValue{}, false, fmt.Errorf("error getting metrics from Datadog: %w", err)
-	}
+	var metric external_metrics.ExternalMetricValue
+	var num float64
+	var err error
+
+	if s.useClusterAgentProxy {
+		url := buildMetricURL(s.metadata.datadogMetricServiceURL, s.metadata.datadogMetricNamespace, s.metadata.hpaMetricName)
+
+		req, err := s.getDatadogClusterAgentHTTPRequest(ctx, url)
+		if (err != nil) || (req == nil) {
+			return []external_metrics.ExternalMetricValue{}, false, fmt.Errorf("error generating http request: %w", err)
+		}
 
-	metric := GenerateMetricInMili(metricName, num)
+		num, err = s.getDatadogMetricValue(req)
+		if err != nil {
+			return []external_metrics.ExternalMetricValue{}, false, fmt.Errorf("error getting metric value: %w", err)
+		}
+
+		metric = GenerateMetricInMili(metricName, num)
+		return []external_metrics.ExternalMetricValue{metric}, num > s.metadata.activationTargetValue, nil
+	} else {
+		num, err = s.getQueryResult(ctx)
+		if err != nil {
+			s.logger.Error(err, "error getting metrics from Datadog")
+			return []external_metrics.ExternalMetricValue{}, false, fmt.Errorf("error getting metrics from Datadog: %w", err)
+		}
+
+		metric = GenerateMetricInMili(metricName, num)
+		return []external_metrics.ExternalMetricValue{metric}, num > s.metadata.activationQueryValue, nil
+	}
```

Comment:
let's remove the extra else
```suggestion
	} 
	num, err = s.getQueryResult(ctx)
	if err != nil {
		s.logger.Error(err, "error getting metrics from Datadog")
		return []external_metrics.ExternalMetricValue{}, false, fmt.Errorf("error getting metrics from Datadog: %w", err)
	}

	metric = GenerateMetricInMili(metricName, num)
	return []external_metrics.ExternalMetricValue{metric}, num > s.metadata.activationQueryValue, nil

```

![image](https://github.com/user-attachments/assets/d7ff5ae9-47f3-4905-a1fe-4ac39d076418)